### PR TITLE
feat: add warning message in case there are people on the waiting list

### LINKED ISSUE #2407: Add warning when canceling the events about the waitlist
### Discuss your ideas or share your views: 
When a user leaves an event which has a waiting list, someone else will join the event in their stead, I don't think this is an issue, just warning about it happening should be enough, if there is a waitlist.

https://user-images.githubusercontent.com/88248797/220564704-57d3d355-0e8f-4bd9-9551-76e401f626b5.webm

Including a body, if there is a waitlist in the section below, should do the trick.

https://github.com/freeCodeCamp/chapter/blob/45be96dc812617511c7c0cd33feef0068b872cea/client/src/modules/events/pages/eventPage.tsx#L137-L139

---

To get set up locally, take a look at our [contribution docs](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md). And, if you'd like to discuss more, you can head on over to [Discord](https://discord.gg/QbQd7BpaaH) and say hi, or comment here if you prefer.

### PR DESCRIPTION
**Pull request by Ismailtlem**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #2407

### CODE REVIEW

File: client/src/modules/events/components/Actions.tsx [hunk 1]
```diff
@@ -143,16 +146,22 @@ export const Actions = ({
...
-    const ok = await confirm({
+    const cancelMessage: {
+      title: string;
+      body?: string;
+    } = {
       title: 'Are you sure you want to cancel your attendance?',
-    });
+    };
+    if (waitlist.length > 0) {
+      cancelMessage.body = 'The next person on the list will take your seat';
```
Comment (gikf) [inline client/src/modules/events/components/Actions.tsx:156]:
```suggestion
      cancelMessage.body = 'The next person on the waitlist will take your seat';
```

File: client/src/modules/events/components/Actions.tsx [hunk 2]
```diff
@@ -143,16 +146,22 @@ export const Actions = ({
...
   async function onCancelAttendance() {
-    const ok = await confirm({
+    const cancelMessage: {
+      title: string;
+      body?: string;
+    } = {
       title: 'Are you sure you want to cancel your attendance?',
-    });
+    };
+    if (waitlist.length > 0) {
```
Comment (gikf) [inline client/src/modules/events/components/Actions.tsx:155]:
```suggestion
    if (waitlist.length > 0 && !inviteOnly) {
```
There's one edge case - invite only event, in which case text shouldn't be displayed. For invite only event it's also not guaranteed canceling user is already off the waitlist at this point.

File: client/src/modules/events/components/Actions.tsx [hunk 3]
```diff
@@ -143,16 +146,22 @@ export const Actions = ({
...
+      cancelMessage.body = 'The next person on the list will take your seat';
+    }
+    const ok = await confirm(cancelMessage);
```
Comment (gikf) [inline client/src/modules/events/components/Actions.tsx:158]:
```suggestion
    const ok = await confirm({
      title: 'Are you sure you want to cancel your attendance?',
      ...(waitlist.length > 0 &&
        !inviteOnly && {
          body: 'The next person on the waitlist will take your seat.',
        }),
    });
```
Creating `cancelMessage` with optional `body` property, and then conditionally adding it separately is a bit verbose. What do you think about conditionally including it in this way?

Comment (Ismailtlem) [reply client/src/modules/events/components/Actions.tsx]: Agree, cleaner this way

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

**Pull request merged**